libs/application/template-api-modules/src/lib/modules/shared/api/vehicles/vehicles.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@nestjs/common'
+import { TemplateApiError } from '@island.is/nest/problem'
+import { BaseTemplateApiService } from '../../../base-template-api.service'
+import { coreErrorMessages } from '@island.is/application/core'
+import { TemplateApiModuleActionProps } from '../../../../types'
+import { VehicleSearchApi } from '@island.is/clients/vehicles'
+import { Auth, AuthMiddleware } from '@island.is/auth-nest-tools'
+import { CurrentVehiclesParameters } from '@island.is/application/types'
+
+@Injectable()
+export class VehiclesService extends BaseTemplateApiService {
+  constructor(private readonly vehiclesApi: VehicleSearchApi) {
+    super('VehiclesShared')
+  }
+
+  private vehiclesApiWithAuth(auth: Auth) {
+    return this.vehiclesApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  async currentVehicles({
+    auth,
+    params,
+  }: TemplateApiModuleActionProps<CurrentVehiclesParameters>) {
+    const showOwned = params?.showOwned || false
+    const showCoOwned = params?.showCoOwned || false
+    const showOperated = params?.showOperated || false
+
+    const result = await this.vehiclesApiWithAuth(auth).currentVehiclesGet({
+      persidNo: auth.nationalId,
+      showOwned: showOwned,
+      showCoowned: showCoOwned,
+      showOperated: showOperated,
+    })
+
+    // // Validate that user has at least 1 vehicle
+    if (!result || !result.length) {
+      if (showOwned && showCoOwned && !showOperated) {
+        throw new TemplateApiError(
+          {
+            title: coreErrorMessages.vehiclesEmptyListOwner,
+            summary: coreErrorMessages.vehiclesEmptyListOwner,
+          },
+          400,
+        )
+      } else if (showOwned && showCoOwned && !showOperated) {
```

Comment:
Fix lint error. Also not really clear how this error flow with the parameters is suppposed to work. You might want to consider some basic documentation on usage.